core/unix/tls.h

```diff
@@ -184,6 +184,17 @@ read_thread_register(reg_id_t reg)
         ASSERT_NOT_REACHED();
         return 0;
     }
+#elif defined(RISCV64)
+    ptr_uint_t sel;
+    if (reg == DR_REG_TP) {
+        asm volatile("mv %0, tp" : "=r"(sel));
+    } else if (reg == DR_REG_INVALID) {
```

Comment:
It's a bit confusing to handle DR_REG_INVALID (which is really meant to denote an error) this way. Ideally the caller shouldn't even invoke this routine with this reg.

Reply:
Actually `DR_REG_INVALID` is a valid choice for `read_thread_register()` on RISC-V. The reason is here: https://github.com/DynamoRIO/dynamorio/pull/5561#discussion_r921392505.

More precisely `read_thread_register()` is/should be called with either `TLS_REG_LIB` or `TLS_REG_ALT`.
I use `DR_REG_INVALID` to express that RISC-V has no alternative TLS register. (also `TLS_REG_ALT` is supposed to be X86-only AFAIK). I chose that (following a suggestion from Derek) as it is a more explicit statement than choosing a dummy register and never using it.

Reply:
To add some context: AArch64 uses [`DR_REG_TPIDRRO_EL0`](https://developer.arm.com/documentation/ddi0601/2020-12/AArch64-Registers/TPIDRRO-EL0--EL0-Read-Only-Software-Thread-ID-Register) for `TLS_REG_ALT` which is a read-only register for userspace (EL0) that may hold thread information for the kernel software (EL1).

Reply:
Okay.